File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.js
// Last updated: 10/22/2025, 12:09:26 AM
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let currSum = 0
    let ans = 0
    for(let i = 0;i < k;i++){
        currSum+=nums[i]
    }
    
    ans = currSum / k
    
    for(i = k; i < nums.length;i++){
        currSum+=nums[i]
        currSum-=nums[i - k]
        ans = Math.max(ans,currSum/k)
    }
    return ans
    
};

File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.py
# Last updated: 10/22/2025, 12:09:28 AM

class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        windowSum = 0
        maxSum = float('-inf')
        windowStart = 0
        
        for windowEnd in range(len(nums)):
            windowSum += nums[windowEnd]
            
            if windowEnd >= k - 1:
                
                maxSum = max(maxSum, (windowSum/k))
                windowSum -= nums[windowStart]
                windowStart += 1
                
        return maxSum
            